File: content/courses/cpp-introduction/codes/twod.cxx
/*
 * twod.cxx
 */ 

#include <iostream>
using namespace std;

int main() {

  float **A;
  int nRows=10;
  int nCols=12;
  A=new float*[nRows];
  for (int n=0;n<nRows;++n){
    A[n]=new float[nCols];
  }
  
  for (int i=0;i<nRows;++i) {
      for (int j=0;j<nCols;++j) {
           A[i][j]=i+j;
      }
  }

  cout<<A[3][4]<<endl;

  return 0;
}

File: content/courses/parallel-computing-introduction/codes/twod.py
import sys
import numpy as np

nrl=4
ncl=4

nrows=nrl+2
ncols=ncl+2
u=np.zeros((nrows,ncols),dtype='float')

for i in range(nrows):
    for j in range(ncols):
        u[i,j]=i+j*2

buf=np.zeros(nrows)
for i in range(nrows):
    buf[i]=u[i%ncols,0]

#Make sure it works

for i in range(nrows):
    for j in range(ncols):
        sys.stdout.write(str(u[i,j])+" ")
    sys.stdout.write("\n")

print("Buffer")
for i in range(nrows):
    sys.stdout.write(str(buf[i])+" ")
sys.stdout.write("\n")
